src/Voron/Data/CompactTrees/CompactTree.cs

```diff
@@ -2057,8 +2057,8 @@ private void SearchInCurrentPage(EncodedKey key, ref CursorState state)
                 if (match >= 0)
                 {
                     bot += mid;
-                    if (match == 0)
-                        goto IsMatch;
+                    if (match == 0 && keyLength == encodedKeyLength)
+                        goto IsExactMatch;
                 }
```

Comment:
@ayende @maciejaszyk I need a review here... Is it really a bug? Does it trigger only after doing the change to use encoded keys? How didn't we trigger this before?

Reply:
Pretty sure that `match == 0` can happen _only_ if the length matches

Reply:
No, we are doing `Math.Min(keyLength, encodedKeyLength)` therefore they can still match but one been bigger than the other.  Which makes me wonder why we have coded it like this, that should be handled by `AdvMemory.CompareInline`. Am I right?

Reply:
Yes, this is needed, see here:
https://github.com/ravendb/ravendb/blob/599f19a74b39f3f2751f93df5c9f0a2373f5abf6/src/Voron/Data/CompactTrees/CompactTree.cs#L2061

We are checking the prefix for the start, then we compare the size.
However... if we have two values that are prefix of one another, and we may find the first one (that isn't a match) and then miss the other one that is a match

Reply:
Yes, that is exactly what I believe it was happening.